src/python/engagement-creator/src/engagement-creator.py

```diff
@@ -206,92 +211,109 @@ def nodes_to_attach_risk_to(
     return [node for node in nodes if node.node_key in risky_node_keys_set]
 
 
+def get_metrics() -> EngagementCreatorMetrics:
+    return EngagementCreatorMetrics(SERVICE_NAME)
+
+
 def lambda_handler(s3_event: S3Event, context: Any) -> None:
     mg_client = MasterGraphClient()
-
     s3 = get_s3_client()
+    metrics = get_metrics()
+
     for event in s3_event["Records"]:
-        if not IS_LOCAL:
-            event = json.loads(event["body"])["Records"][0]
+        with metrics.time_to_process_event():
+            _process_one_event(event, s3, mg_client, metrics)
```

Comment:
If an exception is thrown in `_process_one_event(..)` we'll record a duration but not an "events_processed" count. This seems like the right thing to me as "events_processed" are only the successfully processed events, but should we also count failures? The use case I'm thinking of is that if error handling is very slow, so a block of errors bogs down processing, it would be nice to be able to correlate the latency increase with a failure spike.

Reply:
Sure, makes sense. I'll do

```
            try:
                _process_one_event(event, s3, mg_client)
            except:
                metrics.event_processed(status="failure")
                raise
            else:
                metrics.event_processed(status="success")
```